forge-gui/res/cardsfolder/upcoming/invasion_of_new_phyrexia_teferi_akosa_of_zhalfir.txt

```diff
@@ -0,0 +1,31 @@
+Name:Invasion of New Phyrexia
+ManaCost:X W U
+Types:Battle Siege
+Defense:6
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, create X 2/2 white and blue Knight creature tokens with vigilance.
+SVar:TrigToken:DB$ Token | TokenAmount$ Count$xPaid | TokenScript$ wu_2_2_knight_vigilance
```

Comment:
This feels wrong. I think you have to asssign xPaid to an SVar

Reply:
I think something changed so we don't have to now... @tool4ever will certainly be able to explain better

Reply:
I have no problem changing it, but I'd definitely like to better understand when we need to assign SVars to a line and when we can just nest the calculation in an effect line.

Reply:
X is kind of a special case because it's still pretty hardcoded on lots of AI apis :/